File: shared/dtos/websearch.py
from typing import Literal

from pydantic import BaseModel


class WebSearchResultDto(BaseModel):
    title: str
    url: str
    snippet: str


class WebFetchResultDto(BaseModel):
    url: str
    title: str | None = None
    content: str


class WebSearchProviderDto(BaseModel):
    provider_id: str
    display_name: str
    is_configured: bool
    last_test_status: Literal["untested", "valid", "failed"] | None = None
    last_test_error: str | None = None


class WebSearchCredentialDto(BaseModel):
    provider_id: str
    is_configured: bool
    last_test_status: Literal["untested", "valid", "failed"] | None = None
    last_test_error: str | None = None
    last_test_at: str | None = None


class SetWebSearchKeyDto(BaseModel):
    api_key: str

File: frontend/src/core/types/websearch.ts
export interface WebSearchProvider {
  provider_id: string
  display_name: string
  is_configured: boolean
  last_test_status: 'untested' | 'valid' | 'failed' | null
  last_test_error: string | null
}

export interface WebSearchCredential {
  provider_id: string
  is_configured: boolean
  last_test_status: 'untested' | 'valid' | 'failed' | null
  last_test_error: string | null
  last_test_at: string | null
}

export interface SetWebSearchKeyRequest {
  api_key: string
}

export interface TestWebSearchResponse {
  valid: boolean
  error: string | null
}
